### PR TITLE
fix: CI exception

### PR DESCRIPTION
**Pull request by liuxingbaoyu**
| Q                        | A 
| ------------------------ | ---
| License                  | MIT

1. Fixed CI exception introduced in just merged PR.
2. Fixed the warehouse coverage test exception I introduced earlier. :)
3. Fixed a small bug in the development source map.
4. Make makejs more beautiful.

The main branch CI still has exceptions, but not what we just introduced.
https://github.com/liuxingbaoyu/babel/runs/8099881906?check_suite_focus=true

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/14892"></a>

### CODE REVIEW

File: Makefile
Comment (JLHwung) [inline Makefile:148]: When `BABEL_COVERAGE` is true, we will build Babel with `babel-plugin-istanbul` and the coverage reporter should have worked already. I am not sure whether pulling `--coverage` will transform the instrumented code again.

Comment (liuxingbaoyu) [reply Makefile:148]: https://github.com/babel/babel/pull/14802/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R146-R148

This was a regression I introduced that resulted in our repo not uploading new coverage reports for a month. 🤦‍♂️

**Comment by liuxingbaoyu** [comment]
Prettier, I pulled their repository locally and there is no exception, which is very strange.

File: Makefile
Comment (JLHwung) [reply Makefile:148]: Ok. Good to see the test coverage is 93% now.

Comment (liuxingbaoyu) [reply Makefile:148]: In fact, I have a local branch that improves `light-jest-runner` and adds sub-process coverage, but it seems to have encountered some problems with source map merging, which made it impossible to officially open PR.

Comment (liuxingbaoyu) [reply Makefile:148]: It's been a long time, maybe I should dig into the source map when I have time.🙂

**Comment by nicolo-ribaudo** [comment]
We can probably just skip that failing test for now, since it's hard for the prettier maintainer to consider Babel's breaking changes until we start publishing Babel 8 prereleases.

**Comment by fisker** [comment]
That's definitely bug in Prettier, I'll take a look.

**Comment by fisker** [comment]
Ah that's breaking change to the visitor keys. We are refusing to process non-node values to prevent
 bugs. How do I get the same result? Use `BABEL_8_BREAKING`?

**Comment by liuxingbaoyu** [comment]
@fisker Thanks for your investigation! You can download this and use `BABEL_8_BREAKING`.
https://github.com/babel/babel/suites/8056546495/artifacts/346879282

**Comment by fisker** [comment]
I didn't see any differences with/without `BABEL_8_BREAKING`. But the `VISTOR_KEYS` imported from `@babel/types` are different from v7.18.14 to 7.18.13

```diff
+  "Noop": [],
```

`TSCallSignatureDeclaration` `TSConstructSignatureDeclaration` `TSMethodSignature`   `TSFunctionType` `TSConstructorType` has same diff

```diff
-  "parameters",
+  "params",
-   "typeAnnotation"
+   "returnType"
```

Are these changes expected?

**Comment by liuxingbaoyu** [comment]
https://github.com/babel/babel/pull/13709
Yeah, this should be babel8.

**Comment by fisker** [comment]
How to unblock this?

1. Add new keys to [this file](https://github.com/prettier/prettier/blob/next/src/language-js/traverse/visitor-keys.evaluate.js)
2. Use [`NODE_ENV`](https://github.com/prettier/prettier/blob/40af3ed1ca78986bd56b83321365ea2ec6eb0302/src/main/create-print-pre-check-function.js#L4) to skip the test

**Comment by fisker** [comment]
BTW, babel missed this https://github.com/prettier/prettier/blob/40af3ed1ca78986bd56b83321365ea2ec6eb0302/src/language-js/traverse/visitor-keys.evaluate.js#L25

**Comment by liuxingbaoyu** [comment]
> How to unblock this?
> 
> 1. Add new keys to [this file](https://github.com/prettier/prettier/blob/next/src/language-js/traverse/visitor-keys.evaluate.js?rgh-link-date=2022-08-31T06%3A36%3A34Z)
> 2. Use [`NODE_ENV`](https://github.com/prettier/prettier/blob/40af3ed1ca78986bd56b83321365ea2ec6eb0302/src/main/create-print-pre-check-function.js#L4) to skip the test

It's all good for me.

> BTW, babel missed this `Program.interpreter` [prettier/prettier@`40af3ed`/src/language-js/traverse/visitor-keys.evaluate.js#L25](https://github.com/prettier/prettier/blob/40af3ed1ca78986bd56b83321365ea2ec6eb0302/src/language-js/traverse/visitor-keys.evaluate.js#L25)

I'll take a look when I wake up, I'm even broken now with vscode and git.😓

**Comment by liuxingbaoyu** [comment]
`NODE_ENV=production` seems to introduce new exceptions.😕

**Comment by fisker** [comment]
Sorry forgot about that... it will trigger production test instead of dev test.

**Comment by liuxingbaoyu** [comment]
I turned off this detection manually and it looks fine, thanks again for your help!

**Comment by fisker** [comment]
Good  idea!

`echo "export default () => () => {}" > src/main/create-print-pre-check-function.js`

easier?

File: scripts/integration-tests/e2e-prettier.sh
Comment (JLHwung) [inline scripts/integration-tests/e2e-prettier.sh:39]:
```suggestion
# https://github.com/babel/babel/pull/14892#issuecomment-1233180626
```

**Review by JLHwung — APPROVED** [review]

**Comment by nicolo-ribaudo** [comment]
> Should we start publishing Babel 8 nightly builds?

Yes! I'm a bit busy with university this week, but it's on my list of things to do :smile:

File: scripts/integration-tests/e2e-prettier.sh
Comment (nicolo-ribaudo) [inline scripts/integration-tests/e2e-prettier.sh:39]:
```suggestion
# https://github.com/babel/babel/pull/14892#issuecomment-1233180626
```

File: scripts/integration-tests/e2e-prettier.sh
Comment (liuxingbaoyu) [inline scripts/integration-tests/e2e-prettier.sh:39]: I'm surprised google translate breaks the link.😅

File: jest.config.js
Comment (nicolo-ribaudo) [inline jest.config.js:16]: I don't like that this deletes source maps just because I need to run the tests, but it's good enough for now.

Comment (liuxingbaoyu) [reply jest.config.js:16]: Yes, I hate that too. But I've tried a few things and this seems to work best.😓

**Pull request merged**
